### PR TITLE
Crash fix and simpler graph

### PR DESCRIPTION
**Pull request by nouiz**
This make the graph smaller, so it could speed it up too.

### CODE REVIEW

**Comment by nouiz** [comment]
I added a crash fix.

File: theano/sandbox/cuda/opt.py [hunk 1]
```diff
@@ -314,7 +314,7 @@ def local_gpu_split(node):
             split_node = host_input.owner
             new_op = GpuSplit(split_node.op.len_splits)
             return [new_op(gpu_from_host(split_node.inputs[0]),
-                           *split_node.inputs[1:])[host_input.index]]
+                           *split_node.inputs[1:])]
```
Comment (abergeron) [inline theano/sandbox/cuda/opt.py]: Can you explain this change?  Is it just that the code was blatantly wrong before?

**Comment by abergeron** [comment]
Apart from my confusion I am ok with merging when travis passes.

File: theano/sandbox/cuda/opt.py (hunk 1 above)
Comment (nouiz) [inline theano/sandbox/cuda/opt.py]: There is not direct test of that case. It is hard do that that. The old code was blatently wrong. It always returneded a subtensor of the answer we needed. So it was alwayrs returning an error. I tested the case that give me this problem and the error message disapeared.

Comment (abergeron) [inline theano/sandbox/cuda/opt.py]: Ok then.

**Pull request merged**